providers/base/bin/lid_close_suspend_open.sh

```diff
@@ -1,4 +1,21 @@
-#! /usr/bin/bash
+#!/usr/bin/bash
+
+holdoff_timeout_usec=$(gdbus introspect --system --dest org.freedesktop.login1 --object-path /org/freedesktop/login1 -p | grep HoldoffTimeoutUSec | awk '{print $5}' | awk -F\; '{print $1}')
+holdoff_timeout_sec=$(echo "scale=0; $holdoff_timeout_usec / 1000000" | bc)
+
+if [ "$holdoff_timeout_sec" -ne 0 ]; then
+    if (journalctl --since "$holdoff_timeout_sec seconds ago" -b 0 -r | grep "suspend exit" >/dev/null 2>&1); then
+        echo "The system just resume from suspend, the lid event will be hold off on in $holdoff_timeout_sec seconds"
+        echo "Please wait for the new prompt before starting the test."
+    fi
+    while (journalctl --since "$holdoff_timeout_sec seconds ago" -b 0 -r | grep "suspend exit" >/dev/null 2>&1)
+    do
+        echo "waiting... "
+        sleep 3
+    done
```

Comment:
instead of having a while loop to polling the journal logs, I would suggest we simply sleep within a period of time if needed, likes
```
previous_sleep_log=$(journalctl --output=short-unix --since "$holdoff_timeout_sec seconds ago" -b 0 -r | grep "suspend exit")
if [ "$holdoff_timeout_sec" -ne 0 and "$previous_sleep_log" -ne "" ]; then
    # set the previous_sleep_time
    previous_sleep_time=$(echo $previous_sleep_log | awk -F'.' '{ print $1 }')
    # sleep a period of time
    sleep_time=$(( $(date +"%s") - "$previous_sleep_time" ))
    echo "sleep for $sleep_time seconds ..."
    sleep "$sleep_time"
fi
echo "System is ready for suspend test"
```

Reply:
@stanley31huang a new commit was pushed per this advise. Just a little change. Please have a look when you have time. thanks a lot.